File: exercise/php/day01/tests/SantaCommunicatorTest.php
<?php

declare(strict_types = 1);

use Communication\SantaCommunicator;
use Tests\doubles\TestLogger;

const NORTH_POLE = 'North Pole';
const DASHER = 'Dasher';
const NUMBER_OF_DAYS_TO_REST = 2;
const NUMBER_OF_DAYS_BEFORE_CHRISTMAS = 24;

beforeEach(function (): void {
	$this->logger = new TestLogger;
	$this->communicator = new SantaCommunicator(NUMBER_OF_DAYS_TO_REST);
});

it('composes a message', function (): void {
	$communicator = new SantaCommunicator(NUMBER_OF_DAYS_TO_REST);
	$message = $communicator->composeMessage(DASHER, NORTH_POLE, 5, NUMBER_OF_DAYS_BEFORE_CHRISTMAS);

	expect($message)->toBe('Dear Dasher, please return from North Pole in 17 day(s) to be ready and rest before Christmas.');
});

it('detects overdue reindeer', function (): void {
	$overdue = $this->communicator->isOverdue(DASHER, NORTH_POLE, NUMBER_OF_DAYS_BEFORE_CHRISTMAS, NUMBER_OF_DAYS_BEFORE_CHRISTMAS, $this->logger);

	expect($overdue)->toBeTrue()->
		and($this->logger->loggedMessage())->toBe('Overdue for Dasher located North Pole.');
});

it('returns false when no overdue', function (): void {
	$overdue = $this->communicator->isOverdue(DASHER, NORTH_POLE, 21, NUMBER_OF_DAYS_BEFORE_CHRISTMAS, $this->logger);

	expect($overdue)->toBeFalse();
});

File: solution/day01/src/test/java/SantaCommunicatorTest.java
import communication.Configuration;
import communication.Reindeer;
import communication.SantaCommunicator;
import doubles.TestLogger;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

class SantaCommunicatorTest {
    private final int numberOfDaysToRest = 2;
    private final int numberOfDayBeforeChristmas = 24;
    private final TestLogger logger = new TestLogger();
    private SantaCommunicator communicator;

    @BeforeEach
    void setup() {
        this.communicator = new SantaCommunicator(new Configuration(numberOfDaysToRest, numberOfDayBeforeChristmas));
    }

    @Test
    void composeMessage() {
        assertThat(communicator.composeMessage(
                reindeer(5)
        )).isEqualTo("Dear Dasher, please return from North Pole in 17 day(s) to be ready and rest before Christmas.");
    }

    @Test
    void shouldDetectOverdueReindeer() {
        var overdue = communicator.isOverdue(
                reindeer(numberOfDayBeforeChristmas),
                logger);

        assertThat(overdue).isTrue();
        assertThat(logger.getLog())
                .isEqualTo("Overdue for Dasher located North Pole.");
    }

    @Test
    void shouldReturnFalseWhenNoOverdue() {
        assertThat(communicator.isOverdue(
                reindeer(numberOfDayBeforeChristmas - numberOfDaysToRest - 1),
                logger)
        ).isFalse();
    }

    private static Reindeer reindeer(int numbersOfDaysForComingBack) {
        return new Reindeer("Dasher", "North Pole", numbersOfDaysForComingBack);
    }
}